patches/nginx-1.9.15-hook_error_log.patch

```diff
@@ -0,0 +1,86 @@
+diff --git a/src/core/ngx_log.c b/src/core/ngx_log.c
+index 8e9408d..093351c 100644
+--- a/src/core/ngx_log.c
++++ b/src/core/ngx_log.c
+@@ -164,6 +164,13 @@ ngx_log_error_core(ngx_uint_t level, ngx_log_t *log, ngx_err_t err,
+             break;
+         }
+ 
++        if (log->intercept_log_handler && log->enter_error_log == 0)
++        {
++            log->enter_error_log = 1;
++            log->intercept_log_handler(log, level, errstr, p - errstr);
++            log->enter_error_log = 0;
++        }
++
+         if (log->writer) {
+             log->writer(log, level, errstr, p - errstr);
+             goto next;
+diff --git a/src/core/ngx_log.h b/src/core/ngx_log.h
+index 385fee4..2731db8 100644
+--- a/src/core/ngx_log.h
++++ b/src/core/ngx_log.h
+@@ -45,6 +45,8 @@
+ typedef u_char *(*ngx_log_handler_pt) (ngx_log_t *log, u_char *buf, size_t len);
+ typedef void (*ngx_log_writer_pt) (ngx_log_t *log, ngx_uint_t level,
+     u_char *buf, size_t len);
++typedef ngx_int_t (*ngx_log_intercept_pt) (ngx_log_t *log, ngx_uint_t level,
++    u_char *buf, size_t len);
+ 
+ 
+ struct ngx_log_s {
+@@ -58,6 +60,8 @@ struct ngx_log_s {
+     ngx_log_handler_pt   handler;
+     void                *data;
+ 
++    ngx_log_intercept_pt intercept_log_handler;
++
+     ngx_log_writer_pt    writer;
+     void                *wdata;
+ 
+@@ -70,6 +74,8 @@ struct ngx_log_s {
+     char                *action;
+ 
+     ngx_log_t           *next;
++
++    unsigned             enter_error_log:1;
```

Comment:
There is no point in using `:1` here since there's no other fields can fill up the padding it saves. So it will just waste CPU time while saving *no space*.